### ISSUE TITLE
PHP Notice output breaking nginx proxy

### ISSUE THREAD

**Issue by greghaygood** [wplib/wplib-box#33]
I installed the Layers theme and attempted to load it's Customization screen, and the site crashed.  Checking the nginx error.log, it's throwing the below error.  Granted, this appears to be an issue with the third-party code and I can try to fix that. But in the interests of making wplib-box the "Least Problematic" way to get a dev site up, it would be nice to make sure any errors can pass back up to the user's browser instead of having to dig into the VM and it's logs.  I tried messing with fastcgi buffer sizes, but that didn't seem to help.

```
2016/04/13 21:00:36 [error] 1079#0: *91 FastCGI sent in stderr: "PHP message: PHP Notice:  register_sidebar was called <strong>incorrectly</strong>. No <code>id</code> was set in the arguments array for the "Sidebar" sidebar. Defaulting to "sidebar-1". Manually set the <code>id</code> to "sidebar-1" to silence this notice and keep existing sidebar content. Please see <a href="https://codex.wordpress.org/Debugging_in_WordPress">Debugging in WordPress</a> for more information. (This message was added in version 4.2.0.) in /usr/share/nginx/html/wp/wp-includes/functions.php on line 3901
PHP message: PHP Stack trace:
PHP message: PHP   1. {main}() /usr/share/nginx/html/wp/wp-admin/customize.php:0
PHP message: PHP   2. require_once() /usr/share/nginx/html/wp/wp-admin/customize.php:13
PHP message: PHP   3. require_once() /usr/share/nginx/html/wp/wp-admin/admin.php:31
PHP message: PHP   4. require_once() /usr/share/nginx/html/wp/wp-load.php:42
PHP message: PHP   5. require_once() /usr/share/nginx/html/wp-config.php:83
PHP message: PHP   6. do_action() /usr/share/nginx/html/wp/wp-settings.php:393
PHP message: PHP   7. call_user_func_array:{/usr/share/nginx/html/wp/wp-includes/plugin.php:525}() /usr/share/nginx/html/wp/wp-includes/plugin.php:525
PHP message: PHP   8. wp_widgets_init() /usr/share/nginx/html/wp/wp-includes/plugin.php:525
PHP message: PHP   9. do_action() /usr/share/nginx/html/wp/wp-includes/widgets.php:1475
PHP message: PHP  10. call_user_func_array:{/usr/share/nginx/html/wp/wp-includes/plugin.php:525}() /usr/share/nginx/html/wp/wp-includes/plugin.php:525
PHP message: PHP  11. layers_add_sidebars() /usr/share/nginx/html/wp/wp-includes/plugin.php:525
PHP message: PHP  12. register_sidebar() /usr/share/nginx/html/content/themes/layers/functions.php:17
PHP message: PHP  13. _doing_it_wrong() /usr/share/nginx/html/wp/wp-includes/widgets.php:258
PHP message: PHP  14. trigger_error() /usr/share/nginx/html/wp/wp-includes/functions.php:3901
PHP message: PHP Notice:  register_sidebar was called <strong>incorrectly</
2016/04/13 21:00:36 [error] 1079#0: *91 upstream sent too big header while reading response header from upstream, client: 192.168.33.1, server: localhost, request: "GET /wp/wp-admin/customize.php?url=http%3A%2F%2Fwplib.box%2F HTTP/1.1", upstream: "fastcgi://unix:/var/run/php5-fpm.sock:", host: "wplib.box", referrer: "http://wplib.box/"
```

**Comment by mikeschinkel** [comment]
@greghaygood Thanks for the report.

So this might be the case that you used your code instead of ours?  Our code should have included turning on debugging in the `wp-config-local.php`.  Can you check that?  

This was included in that file:

    define( 'WP_DEBUG', true );
    ini_set('display_errors', 1);
    ini_set('display_startup_errors', 1);
    error_reporting(E_ALL);

If that is not the issue then I would definitely like to address it.

BTW, is this the theme?  

- http://www.layerswp.com/

**Comment by greghaygood** [comment]
This is a vanilla install of wplib-box, and the first thing I did to it was to install the Layers plugin.  So my wp-config-local.php matches what you have here.

And yes, that's the theme.  I made no changes since it failed immediately, so you should be able to replicate easily.

**Comment by mikeschinkel** [comment]
@greghaygood Okay, will try to reproduce.

**Comment by mikeschinkel** [comment]
@greghaygood So I cloned `wplib/wplib-box` and downloaded and placed the Layers theme and was not able to get it to crash.  

I then searched for the string `'to silence this notice and keep existing sidebar content'` and found in one place in WordPress core's `register_sidebar()` function. I think searched for everywhere that function is called and in most cases it is called with a partially hardcoded string for `'id'`.  

The only place I could find that was not hardcoding the `id` is a LayersWP function  `register_dynamic_sidebars()` that iterates over `$a_layers_widget->get_dynamic_sidebars()` _(class `Layers_Widgets`)_ Based on the data you have in your `wp_options` table it is _possible_ but even that seems unlikely.

So I uncommented the `if { ... }` to force the error message to trigger and then it did, but the Query Manager plugin caught the error messages and displayed them in a debugging console accessible via a red flashing button on the Admin Bar.

So I am at a loss.  I would love to find out why and how this failed, but I think I might need your test DB and any code that goes beyond just the box as cloned and the theme as downloaded.  Let me know if you want to package that and send to me; you know how to get it to me. 

BTW, one potential is if you have an `add_action( 'register_sidebar', ... )` in your code, if the filter in not returning the correct value.  But even then, the Query Monitor caught the error when I forced it to happen so not sure how to fix...  Help?
